gb/src/rpi_gpio/dma.rs

```diff
@@ -0,0 +1,382 @@
+                let disable_dma_tx_address = &mut *((self.tx_control_block_memory.virtual_address_ptr as *mut DmaControlBlock).add(tx_cb_index + 2));
+                let start_dma_tx_address = &mut *((self.tx_control_block_memory.virtual_address_ptr as *mut DmaControlBlock).add(tx_cb_index + 3));
+
+                rx_control_block.write_nextconbk(self.tx_control_block_memory.bus_address + ((tx_cb_index + 1) * std::mem::size_of::<DmaControlBlock>()) as u32);
+
+                write_volatile((self.dma_data_memory.virtual_address_ptr as *mut u32).add(i), self.tx_control_block_memory.bus_address + (tx_cb_index * std::mem::size_of::<DmaControlBlock>()) as u32);
+
+                set_dma_tx_address.write_ti(Self::DMA_TI_SRC_INC | Self::DMA_TI_DEST_INC | Self::DMA_TI_WAIT_RESP);
+                set_dma_tx_address.write_source_ad(self.dma_data_memory.bus_address + (i as u32 * 4));
+                set_dma_tx_address.write_dest_ad(Self::DMA_DMA0_CB_PHYS_ADDRESS + (self.tx_channel_number as u32 * 0x100) + 4);  // channel control block address register
+                set_dma_tx_address.write_txfr_len(4);
+                set_dma_tx_address.write_nextconbk(self.tx_control_block_memory.bus_address + ((tx_cb_index + 2) * std::mem::size_of::<DmaControlBlock>()) as u32);
+
+
+                disable_dma_tx_address.write_ti(Self::DMA_TI_SRC_INC | Self::DMA_TI_DEST_INC | Self::DMA_TI_WAIT_RESP);
+                disable_dma_tx_address.write_source_ad(self.dma_const_data_memory.bus_address);
+                disable_dma_tx_address.write_dest_ad(Self::DMA_SPI_CS_PHYS_ADDRESS);
+                disable_dma_tx_address.write_txfr_len(4);
+                disable_dma_tx_address.write_nextconbk(self.tx_control_block_memory.bus_address + ((tx_cb_index + 3) * std::mem::size_of::<DmaControlBlock>()) as u32);
+
+                
+                start_dma_tx_address.write_ti(Self::DMA_TI_SRC_INC | Self::DMA_TI_DEST_INC | Self::DMA_TI_WAIT_RESP);
+                start_dma_tx_address.write_source_ad(self.dma_const_data_memory.bus_address + 4);
+                start_dma_tx_address.write_dest_ad(Self::DMA_DMA0_CB_PHYS_ADDRESS + (self.tx_channel_number as u32 * 0x100) as u32);
+                start_dma_tx_address.write_txfr_len(4);
+                start_dma_tx_address.write_nextconbk(self.rx_control_block_memory.bus_address + (i * std::mem::size_of::<DmaControlBlock>()) as u32);
+
+
+                rx_control_block = &mut *((self.rx_control_block_memory.virtual_address_ptr as *mut DmaControlBlock).add(i));
+                rx_control_block.write_ti(Self::dma_ti_permap(rx_peripherial_mapping) | Self::DMA_TI_SRC_DREQ | Self::DMA_TI_DEST_IGNORE);
+                rx_control_block.write_source_ad(rx_physical_destination_address);
+                rx_control_block.write_dest_ad(0);
+                rx_control_block.write_txfr_len(CHUNK_SIZE as u32 - 4);       // without the 4 byte header
+                rx_control_block.write_nextconbk(0);
+            }
```

Comment:
Maybe this block can live in the new function?

since its the same I can maybe configure it and only copy to the buffer it points to